### ISSUE TITLE
PosPaymentLocalApi throws HttpException 408

### ISSUE THREAD

**Issue by vivekadvsol** [Adyen/adyen-dotnet-api-library#385]
Hi,

**Issue:**
When using PosPaymentLocalApi to send a SaleToPOIRequest it throws a 408 exception.

**Expected behavior**
Payment goes through to terminal

**Additional Info**
Pinpad (P400Plus)

I already checked the similar issue [logged](https://github.com/Adyen/adyen-dotnet-api-library/issues/358) but unable to find any solutions. I think, the problem is the program cannot communicate with the terminal as it doesn't trust the https connection. I can confirm I have already installed the certificate to my Trusted Root.

**Exception**
```
exception	{"HTTP Exception timeout"}	Adyen.HttpClient.HttpClientException
		Code	408	Integer

+		Data	{System.Collections.ListDictionaryInternal}	System.Collections.IDictionary {System.Collections.ListDictionaryInternal}
+		Exception	{"The operation has timed out"}	System.Exception {System.Net.WebException}
		HResult	-2146233088	Integer
		HelpLink	Nothing	String
+		InnerException	Nothing	System.Exception
		JsonResponse	Nothing	String
		Message	"HTTP Exception timeout"	String
		ResponseBody	"No response"	String
		Source	"Adyen"	String
		StackTrace	"   at Adyen.HttpClient.HttpUrlConnectionClient.Request(String endpoint, String json, Config config, Boolean isApiKeyRequired, RequestOptions requestOptions)" & vbCrLf & "   at Adyen.Service.ServiceResource.Request(String json, RequestOptions requestOptions)" & vbCrLf & "   at Adyen.Service.PosPaymentLocalApi.TerminalApiLocal(SaleToPOIMessage saleToPoiRequest, EncryptionCredentialDetails encryptionCredentialDetails)" & vbCrLf & "   at AdvantageAdyen.AdvantageAdyen.MakeSales() in C:\Projects\Plugins\AdvAdyen\AdvAdyen.vb:line 112"	String
```

**Code**

```
Private Function MakeSales() As Boolean
       Try

           Dim config As New Adyen.Config()

           config.Endpoint = "https://192.168.0.9:8443/nexo/"
           config.ApplicationName = "RMS"
           config.Environment = Model.Enum.Environment.Test
           config.HttpRequestTimeout = 120
           config.MerchantAccount = "MERCHENACCOUNT"
           config.XApiKey = "MYAPIKEY"

           Dim myClient As New Client(config)

           Dim encryptionCredentialDetails As New EncryptionCredentialDetails()

           encryptionCredentialDetails.AdyenCryptoVersion = 1
           encryptionCredentialDetails.KeyIdentifier = "KEYIDENTIFIER"
           encryptionCredentialDetails.Password = "PASSWORD"
           encryptionCredentialDetails.KeyVersion = 1

           Dim posPaymentLocalApi As New Adyen.Service.PosPaymentLocalApi(myClient)

           '1.SaleToPOIRequest
           Dim myRequest As New Model.Nexo.Message.SaleToPOIRequest()

           '2.Header
           Dim myReq_MessageHeader As New Model.Nexo.MessageHeader()

           'Load Header
           myReq_MessageHeader.MessageClass = MessageClassType.Service
           myReq_MessageHeader.MessageCategory = MessageCategoryType.Payment
           myReq_MessageHeader.MessageType = MessageType.Request
           myReq_MessageHeader.SaleID = "Test01"
           myReq_MessageHeader.ServiceID = "R129011018"
           myReq_MessageHeader.POIID = "P400-XXXXXXXX"

           '3.Payload
           Dim myPayRequest_PayLoad As New Model.Nexo.PaymentRequest() 'For Body

           '3a.Sale Data & Transaction Id
           Dim myReq_SalesData As New Model.Nexo.SaleData()
           Dim myReq_TransactionID As New Model.Nexo.TransactionIdentification()
           myReq_TransactionID.TransactionID = "Reg012901211018"
           myReq_TransactionID.TimeStamp = DateTime.Now.Date
           myReq_SalesData.SaleTransactionID = myReq_TransactionID

           '3b.PaymentTransaction and 'Amount Req
           Dim myReq_PaymentTransaction As New Model.Nexo.PaymentTransaction()
           Dim myReq_Amount As New Model.Nexo.AmountsReq()

           myReq_Amount.Currency = "AUD"
           myReq_Amount.RequestedAmount = 9.99
           myReq_PaymentTransaction.AmountsReq = myReq_Amount

           '4. Attach Sale Data and Payment Transaction to Payload / PaymentRequest
           myPayRequest_PayLoad.SaleData = myReq_SalesData
           myPayRequest_PayLoad.PaymentTransaction = myReq_PaymentTransaction

           myRequest.MessageHeader = myReq_MessageHeader 'Header
           myRequest.MessagePayload = myPayRequest_PayLoad 'Body
           myRequest.SecurityTrailer = New ContentInformation()

           Dim myResponse As New Model.Nexo.SaleToPOIResponse()

           '5. validate Terminal (returns true)
           TerminalCommonNameValidator.ValidateCertificate("CN=p400-275455552.test.terminal.adyen.com", Model.Enum.Environment.Test)

            '6. Make the request
            myResponse = posPaymentLocalApi.TerminalApiLocal(myRequest, encryptionCredentialDetails)
            MessageBox.Show(myResponse.MessagePayload.ToString())

           Return True

       Catch ex As Exception
           MessageBox.Show(ex.Message)
           Throw New Exception($"Error: {ex}")
       End Try

   End Function
```

**Comment by AlexandrosMor** [comment]
Hello @vivekadvsol,

Thank you for raising this issue. Looks like there is no response as the terminal is unreachable. Some things to try:
1. Try to ping the terminal from the dev enviroment.
2. Try to reach the terminal from postman 
3. Check the the exception message in HttpClientException object

Regards,
Alexandros

**Comment by vivekadvsol** [comment]
Hi Alex,


  1.  I can ping the IP 192.168.0.9
  2.  Postman results are as follows, ( I firmly believe I got the correct api key)
[Graphical user interface, application  Description automatically generated]

[Graphical user interface, application  Description automatically generated]


  1.  Http exception is same as I descripted earlier. Code:408, Exception: The operation has timed out, Message: HTTP Exception timeout.

[Graphical user interface, text, application, email  Description automatically generated]

Regards,

Vivek Rangasamy
Advantage Solutions
T: +61 3 9999 7949
E:  vivekr@advantagesolutions.com.au
W: www.advantage.solution<http://www.advantagesolutions.com.au/>s
